### PR TITLE
1326: decimal grades on the learner record

### PR DESCRIPTION
**Pull request by collinpreston**
#### Pre-Flight checklist

- [x] Screenshots and design review for any changes that affect layout or styling
  - [x] Desktop screenshots
  - [x] Mobile width screenshots
- [x] Testing
  - [x] Code is tested
  - [x] Changes have been manually tested

#### What are the relevant tickets?
https://github.com/mitodl/mitxonline/issues/1326

#### What's this PR do?
Sets the grade returned from the `LearnerRecordSerializer` to have only 2 decimal places.  That serializer is used on the program records page (http://mitxonline.odl.local:8013/records/1/)

#### How should this be manually tested?

1. Have `mitxonline` up and running locally.
2. Set `ENABLE_LEARNER_RECORDS=True` in the .env file.
3. Have a user enrolled into a course run for a course that is a requirement of a program.  Create a course run grade for the course run, and set the grade value as a repeating decimal.
4. Create a partner school record through Django Admin.
5. Navigate to the `mitxonline` Dashboard (http://mitxonline.odl.local:8013/dashboard/?enable_programs&enable_lr).
6. Go to the programs tab and open the program drawer by clicking on the program card.
7. Click "view program record" from the drawer.
8. Ensure that the grade shown for the course is only showing 2 decimal places and not the repeating decimal.

#### Screenshots (if appropriate)
Screenshot 2023-01-09 at 15 35 08

### CODE REVIEW

**Review by jkachel — CHANGES_REQUESTED** [review]
FWIW tested by giving myself a grade in mitxonline of 0.5467666 and still got a rounding error on the program record.

File: courses/serializers.py [hunk 1]
```diff
@@ -760,6 +760,7 @@ def to_representation(self, instance):
                 .order_by("-grade")
                 .first()
             )
+            grade.grade = round(grade.grade, 2)
```
[the remark this answers is not in the document]
Comment (collinpreston) [reply courses/serializers.py:763]: Ha, I used .3333333 and that worked.  Thanks for trying something different and catching that!

File: courses/models.py [hunk 2]
```diff
@@ -1119,7 +1119,7 @@ def to_dict(self):
     def grade_percent(self):
         """Returns the grade field value as a number out of 100 (or None if the value is None)"""
         return (
-            Decimal(self.grade * 100, DecimalContext(prec=2, rounding=ROUND_HALF_EVEN))
+            Decimal(self.grade * 100).quantize(exp=Decimal(1), rounding=ROUND_HALF_EVEN)
```
Comment (collinpreston) [inline courses/models.py:1122]: @jkachel, do you think we should still stick with `ROUND_HALF_EVEN` (ref: https://docs.python.org/3/library/decimal.html#rounding-modes).  I would think we would want something like `ROUND_HALF_UP`.

Comment (jkachel) [reply courses/models.py:1122]: I was a bit conflicted about this when I added the rounding in.. half even is more "fair" over a long timescale (across all grades recorded) but less so for an individual learner so half up may be the better choice.

**Review by jkachel — APPROVED** [review]
LGTM 👍

**Pull request merged**